### PR TITLE
fix(atomic): do not overwrite analyticsClientMiddleware

### PR DESCRIPTION
**Pull request by louis-bompart**
The spread merge was removing the augment included by default, mainly for this issue the `coveoAtomicVersion` metadata.
The handling of a custom analytisClientMiddleware is actually handled at the beginning of `getAnalyticsConfig`, e.g.:
https://github.com/coveo/ui-kit/blob/b15f42c19194152801d4435e2518a50dce8308d0/packages/atomic/src/components/insight/atomic-insight-interface/analytics-config.ts#L12-L15

For the testing, I decided to use a `.each` to reuse the existing tests that ran on the 'simple analytics config' by adding another case where there's an analytics middleware.

### CODE REVIEW

File: packages/atomic/src/components/search/atomic-search-interface/analytics-config.spec.ts [hunk 1]
```diff
@@ -14,90 +14,112 @@ describe('analyticsConfig', () => {
...
-  it('it augment analytics payload with facet title, with any type of facet registered to the store', () => {
-    const resultingConfig = getAnalyticsConfig(config, true, store);
-    (
-      ['facets', 'numericFacets', 'dateFacets', 'categoryFacets'] as const
-    ).forEach((typeOfFacet) => {
-      store.registerFacet(typeOfFacet, {
-        facetId: 'some_id',
-        label: () => 'This is a label',
-        element: document.createElement('div'),
+    it('it augment analytics payload with facet title, with any type of facet registered to the store', () => {
```
Comment (nathanlb) [inline packages/atomic/src/components/search/atomic-search-interface/analytics-config.spec.ts:89]:
```suggestion
    it('augments analytics payload with facet title, with any type of facet registered to the store', () => {
```

File: packages/atomic/src/components/search/atomic-search-interface/analytics-config.spec.ts [hunk 2]
```diff
@@ -14,90 +14,112 @@ describe('analyticsConfig', () => {
...
-      customData: {},
-    }) as any;
-    expect(out.customData).toHaveProperty('coveoAtomicVersion');
-  });
+      expect(out.foo).toBe(
+        config.analytics?.analyticsClientMiddleware ? 'bar' : undefined
+      );
+    });
+
+    it('it augment analytics payload with Atomic version', () => {
```
Comment (nathanlb) [inline packages/atomic/src/components/search/atomic-search-interface/analytics-config.spec.ts:81]:
```suggestion
    it('augments analytics payload with Atomic version', () => {
```

File: packages/atomic/src/components/search/atomic-search-interface/analytics-config.spec.ts [hunk 3]
```diff
@@ -14,90 +14,112 @@ describe('analyticsConfig', () => {
...
+          },
+          facetState: [{title: 'some_title', id: 'some_id'}],
+        }) as any;
+        expect(out.customData.facetTitle).toBe('This is a label');
+        expect(out.facetState[0].title).toBe('This is a label');
       });
+    });
 
-      const out = resultingConfig.analyticsClientMiddleware!('an_event', {
+    it('it does not augment analytics payload with a facet title when a facet is unavailable from the store', () => {
```
Comment (nathanlb) [inline packages/atomic/src/components/search/atomic-search-interface/analytics-config.spec.ts:112]:
```suggestion
    it('does not augment analytics payload with a facet title when a facet is unavailable from the store', () => {
```

**Review by nathanlb — APPROVED** [review]
LGTM

**Review by olamothe — APPROVED** [review]

**Review by btaillon — COMMENTED** [review]
Looks like `atomic-recs-interface/analytics-config.ts` has the same issue

**Pull request merged**